src/uwtools/config/formats/ini.py

```diff
@@ -40,20 +36,12 @@ def _load(self, config_file: OptionalPath) -> dict:
 
         :param config_file: Path to config file to load.
         """
-        # The protected _sections method is the most straightforward way to get at the dict
-        # representation of the parse config.
-
         cfg = configparser.ConfigParser()
-        cfg.optionxform = str  # type: ignore
-        sections = cfg._sections  # type: ignore # pylint: disable=protected-access
+        sections = config_sections(cfg)
         with readable(config_file) as f:
             raw = f.read()
-        try:
             cfg.read_string(raw)
-            return dict(sections)
-        except configparser.MissingSectionHeaderError:
-            cfg.read_string("[top]\n" + raw)
-            return dict(sections.get("top"))
+        return sections
```

Comment:
If the `ConfigParser` -> `dict` one-liner I posted elsewhere works, I think a logical flow here might look like
```
cfg = configparser.ConfigParser()
with readable(config_file) as f:
    cfg.read_string(f.read())
return {s: {k: v for k, v in cfg[s].items()} for s in cfg.sections()}
```
Then we don't need `config_sections()` at all.

The implementation in `SHConfig` might be something like
```
cfg = configparser.ConfigParser()
section = "top"
with readable(config_file) as f:
    cfg.read_string(f"[{section}]\n" + f.read())
return {k: v for k, v in cfg[section].items()}
```
I think...

Reply:
So, weirdly pylint has been throwing `unnecessary-comprehension` errors but insisting on `dict(cfg[section].items())` regardless of what I do. I've disabled that check in ini/sh for now - I think its a quite valuable comprehension, pylint, thank you very much

Reply:
It's actually good advice, we should take it:
```
-        return {s: {k: v for k, v in cfg[s].items()} for s in cfg.sections()}
+        return {s: dict(cfg[s].items()) for s in cfg.sections()}
```
With this change, the `unnecessary-comprehension` complaint disappears. I should have written it that way in the first place.

Also, in general, it's better to suppress linter or typechecker complaints in the most targeted way possible, with either a `# pylint: disable=` trailing comment on the problematic line, or a `# pylint: disable-next=` comment as the line immediately preceding the problematic line. Otherwise, the directive suppresses that check for the entire scope it appears in -- which is the entire file if placed in the outermost scope of the module. That can blind us to issues that we would otherwise be warned about.

Reply:
I was oddly having issues with the inline suppression not reading, which is why I had put it up front to see. But not an issue now - looks like pylint was right. I think I was misunderstanding the instruction, somehow I was not doing exactly what it wanted. But thank you, I've got it all fixed now.

Reply:
Also, unsurprisingly, ChatGPT had no idea how to actually deal with it